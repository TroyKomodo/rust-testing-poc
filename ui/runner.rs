#![cfg_attr(all(coverage_nightly, test), feature(coverage_attribute))]
#![cfg_attr(all(coverage_nightly, test), coverage(off))]

use test_runner::{config, harness, Test};

fn main() {
    let tests = std::fs::read_dir("ui/tests")
        .into_iter()
        .flat_map(|dir| dir.into_iter())
        .filter_map(|dir| dir.ok())
        .filter(|dir| dir.file_type().unwrap().is_file())
        .filter(|dir| dir.path().extension().is_some_and(|ext| ext == "rs"))
        .filter_map(|dir| {
            Some(Test {
                name: dir
                    .path()
                    .to_string_lossy()
                    .strip_suffix(".rs")?
                    .replace("/", "::"),
                path: dir.path(),
            })
        })
        .collect();

    harness(tests, config!());
}
