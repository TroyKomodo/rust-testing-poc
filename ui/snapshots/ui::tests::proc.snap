---
source: ui/runner.rs
input_file: ui/tests/proc.rs
snapshot_kind: text
---
exit status: 0
--- stdout 
#![feature(prelude_import)]
#[prelude_import]
use std::prelude::rust_2021::*;
#[macro_use]
extern crate std;
fn my_fn() {
    match 1 + 2 + 3 {
        tmp => {
            {
                ::std::io::_eprint(format_args!("[{0}:{1}:{2}] {3} = {4:#?}\n",
                        "ui/tests/proc.rs", 2u32, 5u32,
                        "proc::magic_macro2! { 1, 2, 3 }", &tmp));
            };
            tmp
        }
    };
}
